hs_labels/models.py

```diff
@@ -0,0 +1,876 @@
+
+        This list eliminates duplicates.
+
+        :return: List of resource objects marked as mine by a user.
+        """
+        return BaseResource.objects.filter(rlabels__rl2url__ulabels=self,
+                                           rlabels__rl2url__kind=LabelCodes.MINE).distinct()
+
+    # TODO: use access control to include accessible resources in this list.
+    @property
+    def resources_of_interest(self):
+        """
+        Get a list of resources the user has marked in any way.
+
+        :return: List of resource objects that a user has marked.
+        """
+        return BaseResource.objects.filter(rlabels__rl2url__ulabels=self).distinct()
+
+    def get_resources_with_label(self, this_label):
+        """
+        Get a list of resources with a specific label
+        """
+        if not isinstance(this_label, basestring):
+            raise HSLUsageException("Label is not text")
+
+        label_string = UserLabels.clean_label(this_label)                 # remove leading and trailing spaces
+
+        return BaseResource.objects.filter(rlabels__rl2url__ulabels=self,
+                                           rlabels__rl2url__kind=LabelCodes.LABEL,
+                                           rlabels__rl2url__label__exact=label_string)\
+                                        .distinct()\
+                                        .order_by('rlabels__rl2url__label')
+    @property
+    def user_labels(self):
+        return UserResourceLabels.objects.values_list('label', flat=True)\
+                                 .filter(ulabels=self, kind=LabelCodes.LABEL)\
+                                 .distinct().order_by('label')
+
+    ######################################
+    # Label a resource
+    ######################################
+    @staticmethod
+    def clean_label(name):
+        label_string = re.sub('/', r'', name)                   # no /'s
+        label_string = label_string.strip()                     # no leading or trailing whitespace
+        label_string = re.sub(r'\s+', r' ', label_string)       # collapse multiple whitespace, including tabs
+        return label_string
+
+    def label_resource(self, this_resource, this_label):
+        """
+        Share a resource with a group
+
+        :param this_resource: Resource to share.
+        :param this_label: Label to assign to resource
+        :return: None
+
+        Users are allowed to label any resource, including resources to which they do not have access.
+        This is not an access control problem because labeling information is private.
+        """
+
+        # check for user error
+        if not isinstance(this_resource, BaseResource):
+            raise HSLUsageException("Target is not a resource")
+        label_resource = this_resource.rlabels
+
+        if not isinstance(this_label, basestring):
+            raise HSLUsageException("Label is not text")
```

Comment:
Effectively the change would make no difference to behavior, but I do think you should lean a little more towards the conventions and best practices of the tools and language you're applying your skills through. Even when they differ from your usual conventions. It is better to remain consistent within a codebase than within yourself, and every ecosystem and language has its own appreciated ways of doing things. I certainly follow much different sorts of best practices when I'm writing Javascript or C#, or even when I'm writing Python code that isn't a Django project. Conventions are often quite localized.